### PR TITLE
Update our Usage of `virtual`/`override`/`final` in the Slice Compiler

### PR DESCRIPTION
**Pull request by InsertCreativityHere**
This is the first phase of low-hanging-cleanup in the C++ code of the Slice compiler.
Figure it's better to split them into more digestible PRs, since it ends up touching lots of code.

### CODE REVIEW

File: cpp/src/Slice/Parser.h [hunk 1]
```diff
@@ -543,26 +543,26 @@ namespace Slice
...
     // ClassDecl
     // ----------------------------------------------------------------------
 
-    class ClassDecl : public virtual Constructed
+    class ClassDecl final : public virtual Constructed
     {
     public:
         ClassDecl(const ContainerPtr&, const std::string&);
-        virtual void destroy();
+        void destroy() override;
```
Comment (pepone) [inline cpp/src/Slice/Parser.h:557]: Shouldn't all these be final, given ClassDecl is itself final?

File: cpp/src/slice2cpp/Gen.cpp [hunk 2]
```diff
@@ -3293,12 +3293,6 @@ Slice::Gen::StreamVisitor::visitClassDefStart(const ClassDefPtr& c)
     return false;
 }
 
-bool
-Slice::Gen::StreamVisitor::visitExceptionStart(const ExceptionPtr&)
-{
-    return true;
-}
```
Comment (pepone) [inline cpp/src/slice2cpp/Gen.cpp:3300]: Why do you remove this method?

**Review by pepone — APPROVED** [review]
Looks good, see comments.

File: cpp/src/Slice/Parser.h (hunk 1 above)
Comment (InsertCreativityHere) [reply cpp/src/Slice/Parser.h:557]: Yes, you're right. An oversight on my part I guess.

File: cpp/src/slice2cpp/Gen.cpp (hunk 2 above)
Comment (InsertCreativityHere) [reply cpp/src/slice2cpp/Gen.cpp:3300]: Fixed.

**Pull request merged**